tests/unit/spec/base/core-datalayer.js

```diff
@@ -0,0 +1,27 @@
+/* For reference read the Jasmine and Sinon docs
+ * Jasmine docs: http://pivotal.github.io/jasmine/
+ * Sinon docs: http://sinonjs.org/docs/
+ */
+
+/* global describe, beforeEach, afterEach, it, expect, sinon, spyOn, jasmine */
+
+describe('core-datalayer.js', function() {
+
+    describe('PageId', function(){
+
+        it('will grab data-gtm-page-id value if present on <html> element', function(){
+            $('html').attr('data-gtm-page-id', 'test');
+
+            expect(Mozilla.Analytics.getPageId()).toBe('test');
+        });
+
+        it('will grab the pathname minus the first directory if no data-gtm-page-id value is present on <html> element', function(){
+            $('html').removeAttr('data-gtm-page-id');
+
+            expect(Mozilla.Analytics.getPageId()).toBe('');
```

Comment:
Hmm, this isn't really testing that the function works as expected. It could also start failing if we decided to change the location/context of where the tests run. In order to make testing a bit easier, I would suggest changing the function slightly so it accepts passing of parameters. Something like this should work:

```javascript
/**
 * Returns page ID used in Event Category for GA events tracked on page
 * @param  {String} id - GTM page ID
 * @return {String} path - URL path name used as a fallback if id
 */
Analytics.getPageId = function(id, path) {
    var pageId = id ? id : $('html').attr('data-gtm-page-id');
    var pathName = path ? path : document.location.pathname;

    return pageId ? pageId : pathName.replace(/\/[^\/]*/, '');
};
```

We could then test it more thoroughly, asserting multiple different paths return as expected.